core/src/components/segment/segment.tsx

```diff
@@ -78,11 +89,261 @@ export class Segment implements ComponentInterface {
     this.emitStyle();
   }
 
-  componentDidLoad() {
+  componentWillLoad() {
+    this.emitStyle();
+  }
+
+  async componentDidLoad() {
     this.updateButtons();
+    this.setCheckedClasses();
+
+    this.gesture = (await import('../../utils/gesture')).createGesture({
+      el: this.el,
+      gestureName: 'segment',
+      gesturePriority: 100,
+      threshold: 0,
+      passive: false,
+      onStart: ev => this.onStart(ev),
+      onMove: ev => this.onMove(ev),
+      onEnd: ev => this.onEnd(ev),
+    });
+    this.disabledChanged();
+
     this.didInit = true;
   }
 
+  onStart(detail: GestureDetail) {
+    this.activate(detail);
+  }
+
+  onMove(detail: GestureDetail) {
+    this.setNextIndex(detail);
+  }
+
+  onEnd(detail: GestureDetail) {
+    this.activated = false;
+
+    this.setNextIndex(detail, true);
+
+    detail.event.preventDefault();
+    detail.event.stopImmediatePropagation();
+
+    const ripple = (detail.event!.target! as HTMLElement).shadowRoot!.querySelector('ion-ripple-effect');
+    if (!ripple) { return; }
+
+    const { x, y } = pointerCoord(detail.event);
+
+    ripple.addRipple(x, y).then(remove => remove());
```

Comment:
So I'm actually trying this on an Android device and the ripple is being applied to the starting element not the ending element.

Steps to repro:

1. Load the Basic test on Android.
2. Start on "Free" and and swipe to "Top".
3. Release finger. Notice that the ripple is applied to "Free" and not "Top".